PostHog/PostHogSessionManager.swift

```diff
@@ -46,7 +46,17 @@ import Foundation
         timeNow - sessionLastTimestamp > sessionChangeThreshold
     }
 
+    private func isiOSNativeSdk() -> Bool {
+        // postHogSdkName will be set to eg posthog-react-native if not
+        postHogSdkName == postHogiOSSdkName
```

Comment:
The session manager is a singleton that is created before the SDK is initiated so in this case the only "safe" way is every time.
Before moving the check from PostHogSDK to here, I computed only once because I knew the new value was already assigned.
So sadly not.

Reply:
Yep, and that's what I want to avoid since the bug is exactly about that, so it's ok to pay the performance cost which is minimal, and if the bug is solved, I can look into making this better.